### PR TITLE
feat(persons-on-events): add person and groups created_at to events schema

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Problem

We added https://github.com/PostHog/posthog/pull/9510 to add the required persons/groups cols on the events table but have now gotten a request to add created_at columns as well.

## Changes

Migration + required schema changes for persons and groups created_at.

**This PR does not** yet touch plugin server ingestion to send these values, just sets up the required scaffolding in CH to actually "receive" these values

## How did you test this code?

Ran manually, existing tests

### CODE REVIEW

**Comment by macobo** [comment]
Interesting that posthog/clickhouse/test/test_schema.py snapshots weren't updated - mind digging into this before merging it?

File: posthog/clickhouse/migrations/0030_created_at_persons_and_groups_on_events.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        sync_execute(ADD_COLUMNS_BASE_SQL.format(table="writable_events", cluster=CLICKHOUSE_CLUSTER))
+        sync_execute(ADD_COLUMNS_BASE_SQL.format(table="sharded_events", cluster=CLICKHOUSE_CLUSTER))
+
+
+operations = [
+    migrations.RunSQL(f"DROP TABLE IF EXISTS events_json_mv ON CLUSTER '{CLICKHOUSE_CLUSTER}'"),
+    migrations.RunSQL(f"DROP TABLE IF EXISTS kafka_events_json ON CLUSTER '{CLICKHOUSE_CLUSTER}'"),
+    migrations.RunPython(add_columns_to_required_tables),
+    migrations.RunSQL(KAFKA_EVENTS_TABLE_JSON_SQL()),
+    migrations.RunSQL(EVENTS_TABLE_JSON_MV_SQL()),
```
Comment (macobo) [inline posthog/clickhouse/migrations/0030_created_at_persons_and_groups_on_events.py:33]: Is the protobuf table removed at this point?

File: posthog/clickhouse/migrations/0030_created_at_persons_and_groups_on_events.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ADD COLUMN IF NOT EXISTS group2_created_at DateTime64,
+ADD COLUMN IF NOT EXISTS group3_created_at DateTime64,
+ADD COLUMN IF NOT EXISTS group4_created_at DateTime64
+"""
+
+
+def add_columns_to_required_tables(_):
+    sync_execute(ADD_COLUMNS_BASE_SQL.format(table="events", cluster=CLICKHOUSE_CLUSTER))
+
+    if clickhouse_is_replicated():
```
Comment (macobo) [inline posthog/clickhouse/migrations/0030_created_at_persons_and_groups_on_events.py:23]: We made the async migration required last release - this isn't needed.

File: posthog/clickhouse/migrations/0030_created_at_persons_and_groups_on_events.py (hunk 1 above)
Comment (yakkomajuri) [reply posthog/clickhouse/migrations/0030_created_at_persons_and_groups_on_events.py:33]: It's not, but if you upgrade we'll by default flip you over to JSON, so no need to update proto.

Haven't yet pulled the plug on proto though so can't delete the tables on a migration as that'd cause some events in flight to be lost.

I think we probably can pull the plug this next release though, I just need to sit down and think about it

**Comment by neilkakkar** [comment]
About the snapshots, looks like our FOSS changes are the culprit:

https://github.com/PostHog/posthog/blob/master/.github/workflows/ci-backend.yml#L272

cc: @EDsCODE , I think it was our PR instead of tests not running causing the snapshot failures.

**Comment by yakkomajuri** [comment]
@neilkakkar or @EDsCODE will either of you pick that up or should I be doing something?

**Review by macobo — APPROVED** [review]

**Comment by neilkakkar** [comment]
sorry, missed your comment, I'm on it!

**Comment by yakkomajuri** [comment]
well this is annoying - merged master and now things are failing

**Comment by neilkakkar** [comment]
Sorry, that was intentional! The right fix would be to add `posthog/clickhouse` to https://github.com/PostHog/posthog/blob/master/.github/workflows/ci-backend.yml#L272 . I missed it in my PR because there were no changes the failing test warned me about there.

**Pull request merged**